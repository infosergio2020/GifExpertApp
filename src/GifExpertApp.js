import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
// import PropTypes from 'prop-types';

const GifExpertApp = ({defaultCategories = []})=> {
    // const [categories, setCategories] = useState(['One Punch'])
    const [categories, setCategories] = useState( defaultCategories );
    return (
            <>
              <h1 className="header-title"> GifExpertApp </h1>
              <AddCategory setCategories={setCategories}/>
              <hr/>
              <ol>
                {
                    categories.map( category => (
                      <GifGrid key={ category } category={category}/>
                    ))
                }
              </ol>
            </>
        )
}
export default GifExpertApp;