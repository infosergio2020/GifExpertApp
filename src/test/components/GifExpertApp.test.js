import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../GifExpertApp';

describe('Pruebas de GifExpertApp', () => {
    const wrapper = shallow(<GifExpertApp/>)
    test('debe de verse correctamente GifExpertApp', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch','DragonBall'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories}/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    })
    

})
