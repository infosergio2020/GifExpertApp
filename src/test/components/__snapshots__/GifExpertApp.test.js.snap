// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1
    className="header-title"
  >
     GifExpertApp 
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="DragonBall"
      key="DragonBall"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas de GifExpertApp debe de verse correctamente GifExpertApp 1`] = `
<Fragment>
  <h1
    className="header-title"
  >
     GifExpertApp 
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
